sdk/ai/azure-ai-projects/azure/ai/projects/aio/operations/_patch.py

```diff
@@ -616,7 +664,16 @@ async def get_connection_string(self) -> str:
             )
 
             if not get_workspace_response.properties.application_insights:
-                raise ResourceNotFoundError("Application Insights resource was not enabled for this Project.")
+                # create an AppInsights for the customer using the resource group and project name
```

Comment:
Please don't add "write" operation within a "read" function. Let's create another function `get_or_create_connection_string`, and then we can discuss with Azure SDK team for whether we keep both 2 functions, or only `get_or_create_connection_string`.